### DISCUSSION TITLE
What companies or projects are using babashka?

### DISCUSSION THREAD

**Discussion by borkdude** [babashka/babashka#1026] [General]
Please share if your company or project is using babashka. You're welcome to provide a little bit of detail about how you are using it well. No strings attached.

This has several purposes:

- Your company might be mentioned in a sheet of a presentation about babashka or sci. A link to a logo would be helpful.
- This issue would be a good way to notify you of any breaking changes in babashka or sci (which we will try to prevent).
- Your company is listed on [this](https://github.com/babashka/babashka/blob/master/doc/companies.md) page.
- It would help me a lot to understand who is using and how this project is being used.

If you would like to sponsor babashka and related projects, that'd be great as well:

https://opencollective.com/babashka
https://github.com/sponsors/borkdude

I'm also available for [commercial support](https://blog.michielborkent.nl/oss-highlights-nov-dec-2021.html#enterprise-support) if you need that.

Similar discussions for:

- [SCI](https://github.com/babashka/sci/discussions/662)
- [clj-kondo](https://github.com/clj-kondo/clj-kondo/discussions/1397)
- [nbb](https://github.com/borkdude/nbb/discussions/91)

**Comment by ikitommi** [comment]
Metosin

Edit by @borkdude, logo here: https://github.com/borkdude/clj-kondo/issues/438#issuecomment-529418785

**Comment by retrogradeorbit** [comment]
Using sci
Epic Castle 
https://epiccastle.io/images/logo.png

**Comment by cldwalker** [comment]
Reify Health
Logo at https://www.reifyhealth.com/

**Comment by jeroenvandijk** [comment]
Adgoji 

We use Babashka in scripts. Sci is used in tooling around Cloudformation and AWS Lambda (in development).

Logo [here](https://drive.google.com/file/d/1VlUhnrodGmZZd8bQl-PFgJPBdJTf_H4M/view)

**Comment by bmillare** [comment]
AWS

I'm trying it out, not using it in production. Can't get it to build on my work VM though, graal 19.3.1 jdk11 is complaining about unsupported features. I need to build it myself because we are using a different version of glibc. I'm just using the standalone jar for now.

**Comment by borkdude** [comment]
@bmillare babashka is currently built with graalvm-jdk8. There is an issue about jdk11 here:

https://github.com/borkdude/babashka/issues/120
and the branch:
https://github.com/borkdude/babashka/tree/jdk11

Do you have a link to the AWS company?

**Comment by bmillare** [comment]
@borkdude I see. I'll try building on jdk8. I work for AWS, specifically ECS. https://aws.amazon.com/ecs/

**Comment by borkdude** [comment]
@bmillare Cool! 

Babashka now also has a Dockerfile: 

https://github.com/borkdude/babashka/blob/master/Dockerfile

That should give you an idea.

It's deployed to Docker hub too:

https://hub.docker.com/r/borkdude/babashka/

**Comment by borkdude** [comment]
NextJournal is using babashka in their notebook images:

> it’s in our base bash environment and I want to use it for all our more complex scripting needs as well as for custom language runtimes
-- @mk on slack

https://nextjournal.com

**Comment by brunchboy** [comment]
Deep Symmetry just adopted sci to build a plugin for generating byte field diagrams in Asciidoctor and Antora documents. I have to hand it to you! I got quite far building my own interpreter based on an EDN parser, and was as far as having special forms like `def` and `let` and `doseq` work, building diagrams like the one linked to in the [project README](https://github.com/Deep-Symmetry/bytefield-svg) when someone pointed me at sci. I am going to miss out on the fun of finishing off some of the other special forms I had sketched out in my head, but I would never have caught up to you, and now I can get back to porting my documents from LaTeX and focusing on my own core features. 🎉

**Comment by borkdude** [comment]
@brunchboy Thank you! I added this project to the list here: https://github.com/borkdude/sci/#why

**Comment by rodolfo42** [comment]
I'm using `babashka` at Nubank quite a lot lately, it's been really helpful. I've heard of a couple of colleagues that use it as well.

I use it for various things, mostly around transforming outputs of some internal cli commands or semi-structured application log lines (we use clojure so most of our log lines are basically edn) and sometimes performing some actions like downloading/printing referenced files, showing some summaries or reducing map structures to a more human-friendly way. Reduces a great deal of toil for me.

Thanks for the project @borkdude!

I guess our logo would be in this page: https://nubank.com.br/en/press/
But my message here is mostly for the shout-out.

**Comment by bn-darindouglass-zz** [comment]
We at Barracuda use babashka for most new scripting tools. Our kube manifest pre-processor started off as a babashka utility and morphed into a JVM clojure environment that uses SCI to process fn and macro calls

Our logo can be found here: https://www.barracuda.com/media

**Comment by liquidz** [comment]
We are using both of babashka and clj-kondo😃 
Thanks for your awesome job!

TOYOKUMO,Inc.
Logo: https://github.com/borkdude/clj-kondo/issues/438#issuecomment-533347900

**Comment by JduPreez** [comment]
[Lifecheq](http://www.lifecheq.co.za) is making use of sci for many of our tools, allowing development to build the tools and platform, and business to extend it and implement their own solutions. 

Thank you so much for sci. It's absolutely fantastic!

**Comment by borkdude** [comment]
Thanks for sharing you all!

To quote the tweet of @JduPreez:

> @borkdude's Small Clojure Interpreter enabled Lifecheq to add scripting to our tools. Now the whole business is learning #Clojure & everyone's scripting their own solutions. This frees up a lot of time from the dev team. Pretty cool if you ask me

<img width="601" alt="Screenshot 2020-07-01 12 44 03" src="https://user-images.githubusercontent.com/284934/86235370-9e17bb80-bb98-11ea-94a2-0e3008291c56.png">

Business people writing Clojure, that IS pretty cool.

**Comment by stevebuik** [comment]
[Nextdoc.io](https://www.nextdoc.io) is using Babashka every day in devops/monitoring. Planning to add sci to our ETL system soon. 

Love your work!

**Comment by lukaszkorecki** [comment]
[EnjoyHQ](https://enjoyhq.con)

We use bb for a couple of things

- ECS stats scraper daemon, runs as a sidecar with each service we deploy
- operations/infrastructure scripts
- YAML/JSON configuration generator

**Comment by eraad** [comment]
https://datil.com

Build scripts!

**Comment by anthony-khong** [comment]
[Zero One Group](https://zero-one-group.com/)

We use Babashka for:
- General scripting such as this [auto version-bump script in Geni](https://github.com/zero-one-group/geni/blob/develop/scripts/version-bumps.clj)
- Ad-hoc data transformations such as transforming a CSV to pretty-printed EDN.
- Automating GitLab using babashka.curl and [GitLab API](https://docs.gitlab.com/ee/api/), such as batch deleting/labelling/moving issues.

**Comment by huahaiy** [comment]
[Datalevin](https://github.com/juji-io/datalevin) native command line shell use sci to execute query/transaction.

**Comment by takaczapka** [comment]
https://www.hi55.uk

scripted tools!

**Comment by borkdude** [comment]
@huahaiy Thanks! @takaczapka Can you tell more about what you are doing? Using babashka or sci? In what way?

**Comment by takaczapka** [comment]
We use babashka to run operation/admin tasks against live system. Usually it involves calling a REST or GraphQL endpoints with payload loaded from a file or typed as arguments. Our thinking is to develop scripts first, and later (but only if they are prove to be useful) convert them to screens in the web portal. So babashka helps us to iterate quickly and decide what is needed in the longer run and what's not.

**Comment by latacora-gabriel** [comment]
At https://www.latacora.com/, we're using sci for a sandboxed environment (and hopefully bb for some util and devops tasks soon). [Logo is here](https://www.latacora.com/images/logo150x150.png). Thanks for all that you do!

**Comment by neilyio** [comment]
I thought I'd share a `bb.edn` that I've been using at work to manage a project that deploys to GitHub Actions in Docker containers. Babashka's new task runner has been an absolute miracle for me. It lets me simply `bb build` and `bb start` to manage complex Docker commands, and `bb update` will commit, push, trigger the Action, and open a page to the running GitHub CI page so I can watch the test. It's also a nice piece of documentation when I completely forget my choices on this project next week. Thank you @borkdude.

<details>
<summary>bb.edn</summary>

```clojure

{:tasks
 {:requires ([cheshire.core :as json]
             [babashka.fs :as fs]
             [clojure.java.shell :refer [sh]]
             [clojure.string :as str])
  :init     (do
              (defn str-quote [s] (str/join s ["'" "'"]))
              (defn str-space [& xs] (str/join " " xs))
              (defn to-json   [m] (str-quote (json/generate-string m)))
              (defn pass-args [& xs]
                (apply str-space (conj (map str-quote *command-line-args*)
                                       (apply str-space xs))))

              (def *root*         (str (fs/canonicalize "../../")))
              (def *dest*         "/github/workspace")
              (def *code*         "/root/monorepo-config")
              (def *pwd*          (-> "pwd" sh :out))
              (def *in-root*      {:dir *root*})
              (def *image-name*   "project/monorepo-config")
              (def *docker-build* (str-space "docker build"
                                             "--tag" *image-name*))
              (def *docker-run*  (str-space
                                  "docker run"
                                  "--rm"
                                  "--volume" (str *root* ":" *code* ":" "ro")
                                  "--workdir" *dest*)))

  echo        (shell *in-root* (pass-args "echo"))

  build       (shell *in-root* (str-space *docker-build*
                                          ".github/actions/monorepo-config"))

  build-debug (shell *in-root* (str-space *docker-build*
                                          "--progress" "plain"
                                          "--no-cache"
                                          ".github/actions/monorepo-config"))

  start       (shell *in-root* (pass-args *docker-run* *image-name*))

  shell       (shell *in-root* (pass-args *docker-run*
                                          "--interactive"
                                          "--tty"
                                          "--entrypoint /bin/bash"
                                          *image-name*))

  commit      (shell *in-root* "git commit -a -m 'testing the workflow'")

  push        (shell *in-root* "git push -u")

  test        (shell *in-root* (str-space "gh workflow run core-ts-check-pr.yml"
                                          "--repo project/project"))

  view        (shell *in-root* "gh run view --web")

  update      (do (run 'commit)
                  (run 'push)
                  (run 'test)
                  (Thread/sleep 2000)
                  (run 'view))}}
```
</details>

**Comment by borkdude** [comment]
@neilyio Thanks a lot for sharing!

**Comment by joelittlejohn** [comment]
We use babashka at JUXT as a general-purpose bash replacement for all kinds of scripting tasks. We've used it to create build and deploy scripts for small projects, and we've used to it create small utilities and more fully featured CLI tools. 

Logo at https://www.juxt.land/logo-normal.svg.

**Comment by borkdude** [comment]
@joelittlejohn Great to hear, thanks for sharing.

**Comment by interstar** [comment]
Using SCI for scripting in Cardigan Bay : 

https://www.youtube.com/watch?v=H7_THeK9EBw

https://github.com/interstar/cardigan-bay/releases/tag/0.6.0

**Comment by borkdude** [comment]
@interstar Thanks for sharing. I added your project here: https://github.com/borkdude/sci#projects-using-sci

**Comment by pmonks** [comment]
I'm using `sci` in this little Discord bot: https://github.com/pmonks/for-science

**Comment by jgrodziski** [comment]
Use SCI for scripting of the [administration tasks of keycloak-clojure](https://cljdoc.org/d/keycloak-clojure/keycloak-clojure/1.18.0/doc/administrative-tasks#realm-setup-with-keycloak-clojure-starter-cli)

**Comment by borkdude** [comment]
@pmonks @jgrodziski Thanks for sharing! Added the projects to the SCI readme.

**Comment by rdgd** [comment]
[Splash Financial](https://www.splashfinancial.com/) uses Babashka for scripting things which Bash just isn't suited for (and because we love Clojure)

[Company logo](https://www.splashfinancial.com/wp-content/uploads/2021/05/logo.svg)

**Comment by borkdude** [comment]
I've created a companies.md page in the babashka repo now:

https://github.com/babashka/babashka/blob/master/doc/companies.md

Sponsoring companies are listed first, in order of payment. They will be able to provide a direct hiring link as well. Remaining companies are listed in alphabetical order. I will create a similar page for clj-kondo and SCI.

In other news, I have quit my long time job to work more on my open source projects. Any additional sponsoring will help me continue doing so! :-)

**Comment by askonomm** [comment]
[Fluent](http://fluent.to), where I work, uses Babashka for tasks. 

[Logo](https://uploads-ssl.webflow.com/600fdbbcf0cbfe02f1a48030/600fdf0fffe881c3e8e298f4_Fluent%20logotype-p-500.png)

**Reply by borkdude** [reply]
Added to the list of sponsoring companies now:

https://github.com/babashka/babashka/blob/master/doc/companies.md#sponsoring-companies

Thanks for sponsoring!

**Comment by stask** [comment]
PractiTest, build scripts and utilities
https://practitest.com

**Comment by borkdude** [comment]
Converted to a discussion so it's easier to ask for details in comments on posts.

**Comment by athos** [comment]
At [Xcoo](https://xcoo.com/), we're using babashka for various kinds of scripting tasks and replacing more and more Bash scripts with it.
Thank you for your efforts in developing this amazing tool!

**Reply by borkdude** [reply]
Thank you, added!

**Comment by hansbugge** [comment]
At [Deon Digital](https://www.deondigital.com) we are using babashka for various CI scripts, and we are using babashka tasks in at least one project.

**Reply by borkdude** [reply]
Thank you, added!

**Comment by nha** [comment]
At [Turtlequeue Ltd](https://turtlequeue.com/) I'm using babashka for all the scripts (CI, build, deployment..). This allowed me to replace a mess of tangled bash scripts that were increasingly relying on environment variables to pass (implicit) context between them. I gave a try to [Mach](https://github.com/juxt/mach) but it is not as pleasant as `bb` (programmers gotta program).

Because all the CI uses `bb` now I made a handy [Github action helper](https://github.com/turtlequeue/setup-babashka)

[logo](https://turtlequeue.com/logo_black.png)

**Reply by borkdude** [reply]
@nha Thank you! We're also using your Github action runner in clj-easy/graal-build-time:

https://github.com/clj-easy/graal-build-time/blob/130819a1d953f5864b8ef3d273997dfa014860c2/.github/workflows/ci.yml#L26

**Comment by n2o** [comment]
Hi, I am one of the founders of [schnaq](https://schnaq.com) ([Logo](https://s3.schnaq.com/schnaq-common/logos/schnaq.svg)), we use babashka currently for additional backup scripts on our clusters, e.g. to upload assets to S3. We are considering babashka for static page generation (as a jekyll replacement).
That's a full Clojure / ClojureScript / babashka company. I am personally sponsoring you and hope that our company will sponsor you also in the future ;-)

**Reply by borkdude** [reply]
Thank you!

**Comment by robert-stuttaford** [comment]
Please do include Cognician :-) Logo: https://avatars.githubusercontent.com/u/1450774?s=200&v=4

**Reply by borkdude** [reply]
Added to https://github.com/babashka/babashka/blob/master/doc/companies.md !

**Comment by borkdude** [comment]
I made a separate discussion for SCI:

https://github.com/babashka/sci/discussions/662

If you are using SCI apart from babashka, please add to that discussion! It helps me to stay informed of how it's being used. Thank you!

**Comment by rafaeldelboni** [comment]
[180s](https://www.180s.com.br/) is using bb for internal clis and pipelines
Logo: https://github.com/180seg

**Reply by borkdude** [reply]
@rafaeldelboni Thanks for becoming a sponsor! I will add the logo to the relevant pages tomorrow.

**Reply by borkdude** [reply]
Added to:

- https://github.com/sponsors/borkdude?preview=true
- https://github.com/babashka/babashka/blob/master/doc/companies.md#180seg

**Comment by cap10morgan** [comment]
[Fluree](https://flur.ee) is using babashka in our upcoming [Nexus](https://flur.ee/nexus/) product in the AWS lambda backend.

Logo: https://i0.wp.com/flur.ee/wp-content/uploads/2021/03/Dark-Horizontal.png?fit=9657%2C3399&ssl=1

**Reply by borkdude** [reply]
Good to hear! I'll add the logo to the companies page tomorrow.

**Comment by keerts** [comment]
[Avisi Apps](https://apps.avisi.com/) is using Babashka for everything that goes beyond what one can comfortably do in a Makefile.

**Reply by borkdude** [reply]
Thanks for letting me know! Hope to visit Avisi again some day at a Clojure meetup perhaps :)

**Reply by keerts** [reply]
We should definitely organize one some day soon!

**Comment by NoahTheDuke** [comment]
[Crossbeam](https://crossbeam.com) uses Babashka for a large portion of our scripting.

**Comment by sogaiu** [comment]
We use Babashka tasks [1] for tree-sitter-clojure development and testing [here](https://github.com/sogaiu/ts-clojure).

We also use Babashka tasks for the associated [clojars-samples](https://github.com/sogaiu/clojars-samples) repository.

---

[1] With a slight extension to our `bb.edn` to support tags for each task that allows us to view tasks by tag using [this utility](https://github.com/sogaiu/babashka-tasks-view).

**Comment by agilepoodle** [comment]
We use babashka for our builds and cloud deployment both in dev and production.

[CarbonLink Oy](https://www.carbonlink.fi/)

**Reply by borkdude** [reply]
Thanks for sharing!
